### PR TITLE
Check query permissions by account in torii endpoint

### LINKED ISSUE #1155: Add permission checks for query endpoint based on signature

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

Closes #1155 

### Description of the Change
Check query permissions by account in torii endpoint

### Benefits
Planned project feature.

### CODE REVIEW

File: iroha/src/smartcontracts/isi/query.rs [hunk 1]
```diff
@@ -2,43 +2,83 @@
 
 use std::{convert::TryFrom, error::Error as StdError, fmt};
 
-use iroha_data_model::prelude::*;
+use iroha_data_model::{prelude::*, query::Payload as QueryPayload};
```
Comment (i1i1) [inline iroha/src/smartcontracts/isi/query.rs:5]: May be just leave `query` here

File: iroha/src/torii.rs [hunk 2]
```diff
@@ -21,25 +21,25 @@ use tokio::{sync::RwLock, task};
...
+    smartcontracts::{isi::query::VerifiedQueryRequest, permissions::IsQueryAllowedBoxed},
     sumeragi::{
         message::VersionedMessage as SumeragiVersionedMessage, GetLeader, IsLeader, SumeragiTrait,
     },
     wsv::WorldTrait,
 };
 
 /// Main network handler and the only entrypoint of the Iroha.
-#[derive(Debug)]
+#[allow(missing_debug_implementations)]
```
Comment (i1i1) [inline iroha/src/torii.rs:36]: May be we can disable this lint, if it doesnt work for generics?

File: iroha/test_network/src/lib.rs [hunk 3]
```diff
@@ -47,7 +47,7 @@ use tokio::{
 struct ShutdownRuntime;
 
 /// Network of peers
-#[derive(Debug)]
+#[allow(missing_debug_implementations)]
```
Comment (i1i1) [inline iroha/test_network/src/lib.rs:50]: Just grepped. We have 14 of such allows

File: iroha_data_model/src/lib.rs [hunk 4]
```diff
@@ -1655,6 +1655,19 @@ pub mod domain {
                 ))
             }
         }
+
+        /// Domain constructor with presetup accounts. Useful for testing purposes.
+        pub fn with_accounts(name: &str, accounts: impl IntoIterator<Item = Account>) -> Self {
+            let mut accounts_map = AccountsMap::new();
```
Comment (i1i1) [inline iroha_data_model/src/lib.rs:1661]: You can collect into `AccountsMap` right? if so, then we can do like this:
```rust
let accounts = accounts.into_iter().collect();
```

**Review by rkharisov — APPROVED** [review]

File: iroha_data_model/src/lib.rs [hunk 5]
```diff
@@ -1655,6 +1655,19 @@ pub mod domain {
...
+            for account in accounts {
+                drop(accounts_map.insert(account.id.clone(), account));
```
Comment (i1i1) [inline iroha_data_model/src/lib.rs:1663]: May be we can discuss somehow `unused_results` lint? I dont feel like we benefiting from it anyhow. Even worse this drop here is not semantically relevant (as drop for mutexguards for example). Besides this drives reader away. Main thing here is that `accounts_map` is getting inserted, not that we drop previous item from where.

File: iroha/src/torii.rs (hunk 2 above)
Comment (eadventurous) [reply iroha/src/torii.rs:36]: Ok

**Review by i1i1 — APPROVED** [review]

**Pull request merged**